access/views.py

```diff
@@ -276,11 +400,24 @@ def children_recursion(parent):
     if errors:
         data["errors"] = errors
 
-    if "gitmanager" in settings.INSTALLED_APPS:
-        data["build_log_url"] = request.build_absolute_uri(reverse("build-log-json", args=(course_key, )))
     return JsonResponse(data)
 
 
+class LoginView(View):
+    def get(self, request):
+        response = render(request, 'access/login.html')
+        response.delete_cookie("AuthToken")
+        return response
+
+    def post(self, request):
+        if not hasattr(request, "user") or not request.user.is_authenticated:
+            return HttpResponse("Invalid token", status=401)
+        else:
+            response = HttpResponse()
+            response.set_cookie("AuthToken", str(request.auth))
```

Comment:
Should the cookie security be improved with parameters `secure=True, httponly=True`? What about expiration time?

https://docs.djangoproject.com/en/3.2/ref/request-response/#django.http.HttpResponse.set_cookie

Reply:
I didn't include expiration time because the token itself has one. The others I added